test/batch.js

```diff
@@ -112,29 +112,31 @@ describe('batch', () => {
   })
 
   describe('with matching enabled', () => {
+    const input = [
+      { id: 'a' },
+      { id: 'b' },
+      { id: 'a' }
+    ]
+
     const asyncListFn = () =>
-      (spy(), Promise.resolve([{ id: 'c' }, { id: 'b' }]))
+      (spy(), Promise.resolve([{ id: 'a', name: 'Spot' }]))
 
     const batched =
-      batch({ inputKey: identity, outputKey: prop('id') }, asyncListFn)
+      batch({ inputKey: prop('id'), outputKey: prop('id') }, asyncListFn)
 
-    const test = evolveP({
-      a: batched,
-      b: batched,
-      c: batched
-    })
+    const test = mapP(batched)
 
     beforeEach(() =>
-      test(obj).then(res)
+      test(input).then(res)
     )
 
-    it('matches each result to the original input', () => {
+    it('matches each result to the original input, accounting for duplicates', () => {
       expect(spy.calls.length).to.equal(1)
-      expect(res()).to.eql({
-        a: undefined,
-        b: { id: 'b' },
-        c: { id: 'c' }
-      })
+      expect(res()).to.eql([
+        { id: 'a', name: 'Spot' },
+        undefined,
+        { id: 'a', name: 'Spot' }
+      ])
```

Comment:
:confused: This test has changed a lot. Should this be an additional test?

Reply:
Sure, I can make a new test.

Reply:
Ok, I've restored the previous test.  Let me know what you think.